server/server.js

```diff
@@ -228,4 +233,154 @@ app.get("/api/feed/:id/:apiId", async (req, res) =>  {
 
 });
 
+
+//add a new comment on Book Page
+app.post("/api/comment", async (req, res) =>  {
+
+    try {
+
+        const {auth0_sub, api_id, text, date, rate } = req.body;
+
+        let user_id = await getUserIdFromSub(auth0_sub)
+
+        const { rows: comment } = await db.query("INSERT INTO comments (user_id, api_id, text, date, rate) VALUES ($1, $2, $3, $4, $5) RETURNING *", [user_id, api_id, text, date, rate]);
+
+        if (comment.length === 0) {
+            res.status(404).json({ message: "Comment not found" });
+        } else {
+            res.status(200).json(comment);
+        }
+
+    } catch (error) {
+        console.error("Error Message!:", error.message);
+        res.status(500).json({ message: error.message });
+    }
+
+});
+
+
+
+//gets the book's comments
+app.get("/api/comment/:id/:userSub", async (req, res) =>  {
+    
+    try {
+        const { id, userSub } = req.params;
+        let loggedInUserId = await getUserIdFromSub(userSub)
+        
+        const {rows : commentList} = await db.query(`
+            SELECT 
+                comments.comment_id, 
+                comments.text, 
+                comments.date, 
+                comments.rate, 
+                users.first_name, 
+                users.last_name,
+                users.auth0_sub
+            FROM comments 
+            JOIN users ON comments.user_id = users.user_id
+            WHERE comments.api_id = $1
+            ORDER BY 
+                CASE WHEN comments.user_id = $2 THEN 0 ELSE 1 END, 
+                comments.date DESC
+        `, [id, loggedInUserId]);
+
+        if (commentList.length === 0) {
+            res.status(200).json([]); 
+        } 
+        else {
+            res.status(200).json(commentList);
+        }
+        
+    } catch (error) {
+        console.error("Error Message!:", error.message);
+        res.status(500).json({ message: error.message });
+    }
+
+});
+
+
+//delete a comment
+app.delete('/api/comment/:id', async (req, res) => {
+    try {
+        const {id} = req.params;
+        const deletePost= await db.query("DELETE FROM comments WHERE comment_id = $1", [id]
+        );
+
+        res.json("The comment was deleted!");
```

Comment:
It's an incredibly rare occasion when a non-object or non-array is returned as a JSON response.

```suggestion
        res.json({ message: "The comment was deleted!"});
```